compiler/p/codegen/PPCInstruction.cpp

```diff
@@ -593,33 +622,76 @@ bool TR::PPCSrc2Instruction::usesRegister(TR::Register *reg)
 
 void TR::PPCSrc2Instruction::assignRegisters(TR_RegisterKinds kindToBeAssigned)
    {
-   TR::Register           *virtualRegister1 = getSource2Register();
-   TR::Register           *virtualRegister2 = getSource1Register();
+   TR::Register           *virtualRegister1 = getSource1Register();
+   TR::Register           *virtualRegister2 = getSource2Register();
    TR::RealRegister       *assignedRegister;
    TR::Machine *machine = cg()->machine();
+   bool excludeGPR0 = getOpCodeValue() == TR::InstOpCode::stxvl;
```

Comment:
Not sure if this is a one off or not but if it is prevalent to exclude gr0 for a variety of different instructions we can consider adding a new instruction property. Food for thought.

Reply:
You're right about it being prevalent. I've already got #5297 open for this exact problem and work is already underway to get that implemented before the lists get any longer than they already are.